packages/hierarchies-react/src/test/internal/TreeLoader.test.ts

```diff
@@ -327,16 +328,20 @@ describe("TreeLoader", () => {
 });
 
 async function collectNodes(loadObs: Observable<LoadedTreePart>) {
-  const nodes = new Map<string | undefined, TreeModelNode[]>();
-  return new Promise<Map<string | undefined, TreeModelNode[]>>((resolve) => {
+  const nodes = new Map<string | undefined, (TreeModelHierarchyNode | ErrorInfo)[]>();
```

Comment:
This isn't great. We should test the library and not our test code. Now we have tests like "loads info node when RowsLimitExceededError is thrown", which basically test this `collectNodes` function. I think this function should return either nodes' array, or an error. And each test should be adjusted to have appropriate name.